src/sparsematrix.jl

```diff
@@ -1969,25 +1969,67 @@ function ==(A1::AbstractSparseMatrixCSC, A2::AbstractSparseMatrixCSC)
                 j2+=1
             else
                 if r1<r2
-                    vals1[j1]!=0 && return false
```

Comment:
all these pesky legacy code.

Reply:
We need traits, defined in a machine-readable way, and a respective checker that flags problems.